### PR TITLE
add validation for user settings page

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: app/elements/common/settings/settings-page.html [hunk 1]
```diff
@@ -41,21 +42,27 @@
...
 					<div>
 						<div class="list-item">
 							<div class="item-label">Name</div>
 							<div>{{fullName}}</div>
 						</div>
 						<div class="aligned-item">
-							<paper-input label="First name" value="{{user.firstName}}"></paper-input>
-							<paper-input label="Last name" value="{{user.lastName}}"></paper-input>
+							<paper-input label="First name" value="{{user.firstName}}"
+								pattern="[a-zA-Z]+" minlength="5" maxlength="40" auto-validate
```
Comment (lorenzleutgeb) [inline app/elements/common/settings/settings-page.html]: `[a-zA-Z]+` is too restrictive. A first name could have 'ä', 'é' and maybe even whitespace.

File: app/elements/common/settings/settings-page.html [hunk 2]
```diff
@@ -41,21 +42,27 @@
...
+								error-message="Must be between 5 and 40 chars and only contain letters."></paper-input>
+							<paper-input label="Last name" value="{{user.lastName}}"
+								pattern="[a-zA-Z]+" minlength="5" maxlength="40" auto-validate
```
Comment (lorenzleutgeb) [inline app/elements/common/settings/settings-page.html]: `[a-zA-Z]+` is too restrictive. A last name could have 'ä', 'é' and maybe even whitespace.

File: app/elements/common/settings/settings-page.html [hunk 3]
```diff
@@ -65,7 +72,8 @@
 					<div>
 						<div class="list-item">
 							<div class="item-label">Email</div>
-							<paper-input value="{{user.email}}" no-label-float></paper-input>
+							<paper-input value="{{user.email}}" no-label-float type="email"
+							required auto-validate error-message="Please enter a valid email 'example@email.com'"></paper-input>
```
Comment (lorenzleutgeb) [inline app/elements/common/settings/settings-page.html]: For examples and docs please use domain names compliant with [RFC 2606](https://tools.ietf.org/html/rfc2606#section-3). "email.com" is a valid domain owned by somebody else. Use "me@example.com" or "me@mail.example" or something similar.

File: app/elements/common/settings/settings-page.html [hunk 4]
```diff
@@ -41,21 +42,27 @@
...
+								pattern="[a-zA-Z]+" minlength="5" maxlength="40" auto-validate
+								error-message="Must be between 5 and 40 chars and only contain letters."></paper-input>
 						</div>
 					</div>
 					<div>
 						<div class="list-item">
 							<div class="item-label">Username</div>
-							<paper-input value="{{user.username}}" no-label-float></paper-input>
+							<paper-input value="{{user.username}}" no-label-float
+								required minlength="5" maxlength="40" auto-validate pattern="^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$"
```
Comment (lorenzleutgeb) [inline app/elements/common/settings/settings-page.html:64]: THis regex allows for a maximum of one dash ("-") in the username. What is the point here?

File: app/elements/common/settings/settings-page.html [hunk 5]
```diff
@@ -98,12 +106,26 @@
...
 		},
+		listeners: {
+			'keyup': '_userChanged'
+		},
 		ready: function(){
 			this.$$('user-service').whoAmI();
 		},
+		_userChanged: function(){
+			this.saveDisabled = !this.$.form.validate();
+			console.log(this.$.form.validate());
```
Comment (lorenzleutgeb) [inline app/elements/common/settings/settings-page.html]: d

File: app/elements/common/settings/settings-page.html (hunk 4 above)
Comment (victorbalan) [inline app/elements/common/settings/settings-page.html:64]: i copied it from platform

**Pull request merged**